### PR TITLE
build: Fix realloc usage

### PR DESCRIPTION
**Pull request by ueno**
As realloc() doesn't touch the original memory block, we need to use a local variable to avoid potential memory leak in failure cases.

Pointed by David Woodhouse in #270.

### CODE REVIEW

File: p11-kit/filter.c [hunk 1]
```diff
@@ -96,11 +96,12 @@ static bool
...
 {
 	if (filter->n_slots >= filter->max_slots) {
+		FilterSlot *slots;
 		filter->max_slots = filter->max_slots * 2 + 1;
-		filter->slots = realloc (filter->slots,
-					 filter->max_slots * sizeof (FilterSlot));
-		if (filter->slots == NULL)
-			return false;
+		slots = realloc (filter->slots,
+				 filter->max_slots * sizeof (FilterSlot));
```
Comment (Jakuje) [inline p11-kit/filter.c]: This line still mixes tabs and spaces for indentation. Not sure what is the proper coding style here though

Comment (ueno) [reply p11-kit/filter.c]: Our coding style is a mixture of the Linux kernel coding style and the GNOME coding style: the function arguments are aligned to the left parenthesis, while the basic indentation is a hard tab. Maybe we should document it clearly in somewhere.

**Comment by ueno** [comment]
Thanks for the review!

File: p11-kit/filter.c (hunk 1 above)
Comment (Jakuje) [reply p11-kit/filter.c]: This aligns nicely in the diff in email, but for some reason does not in this view from github (it renders for some reason tabs in 67 spaces). To fix this for good everywhere, the two tabs would have to be rewritten to spaces, but it is no big deal.

**Pull request merged**